.github/workflows/ci.yaml

```diff
@@ -1,9 +1,11 @@
-name: CI
-on: [push]
+name: ci
+on:
+  push:
+    branches-ignore: [master, main]
 jobs:
-  validate:
+  build:
     runs-on: ubuntu-latest
-    timeout-minutes: 10
+    timeout-minutes: 5
```

Comment:
The `ci` and `pull-request` workflow runs have all taken under 2 minutes (and some under 1 minute); mainly setting this at 5 to avoid a runaway process. 😉